### PR TITLE
[secure-random-password] Update definition

### PR DESCRIPTION
**Pull request by GaspardFRDev**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mkropat/secure-random-password
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/secure-random-password/index.d.ts
Comment (amcasey) [inline types/secure-random-password/index.d.ts:8]: Personally, I find the alphabetized list less clear.

File: types/secure-random-password/index.d.ts
Comment (amcasey) [inline types/secure-random-password/index.d.ts:23]: It can be an array of `RandomPasswordCharacterSet` but not an individual one?

**Review by amcasey — CHANGES_REQUESTED** [review]
Minor questions.

File: types/secure-random-password/index.d.ts
Comment (GaspardFRDev) [reply types/secure-random-password/index.d.ts:23]: It seems to work with an individual one but it's more straightforward to do : 
`randomPassword({ characters: srp.upper, length: 5})` 
instead of 
`randomPassword({ characters: { characters: srp.upper }, length: 5}})` 
altough you can still use one with array :
`randomPassword({ characters: [{ characters: srp.upper }], length: 5}})`

File: types/secure-random-password/index.d.ts
Comment (GaspardFRDev) [reply types/secure-random-password/index.d.ts:8]: Ok, I can simply add space between each categorie then ?
```
export const lower: string;
export const upper: string;

export const consonants: string;
export const vowels: string;
export const digits: string;

export const symbols: string;
export const copyableSymbols: string;
export const fullSymbols: string;

```

File: types/secure-random-password/index.d.ts
Comment (GaspardFRDev) [inline types/secure-random-password/index.d.ts:18]: I'm wondering if I can restrict exactly use in order to forbid this case which isn't allowed :
```
randomPassword({  characters: 
  [
    { characters: srp.symbols, exactly: 1 },
  ]
});
```

File: types/secure-random-password/index.d.ts
Comment (amcasey) [reply types/secure-random-password/index.d.ts:8]: Works for me.

File: types/secure-random-password/index.d.ts
Comment (amcasey) [reply types/secure-random-password/index.d.ts:23]: I would lean towards matching the underlying library in this case (i.e. by allowing an individual `RandomPasswordCharacterSet`), but I don't feel strongly about it.

File: types/secure-random-password/index.d.ts
Comment (amcasey) [reply types/secure-random-password/index.d.ts:18]: Generally, speaking, it's hard to prevent calls from providing extra properties (e.g. an `{x: number, y:string}` is usable as an `{x:number}`), but you could probably limit it for literals (where unexpected properties are flagged).  Having said that, I'm not sure what you're trying to prevent.

**Review by amcasey — APPROVED** [review]
I don't have any concerns that would block merging, so I'll leave it to you to decide when you're ready.

**Review by amcasey — APPROVED** [review]

**Comment by GaspardFRDev** [comment]
Ready to merge

**Pull request merged**
